haystack/components/agents/agent.py

```diff
@@ -245,6 +255,9 @@ def run(
         state.set("messages", messages)
 
         generator_inputs = self._prepare_generator_inputs(streaming_callback=streaming_callback)
+        streaming_callback = select_streaming_callback(
+            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=False
+        )
```

Comment:
Could we move this to `_prepare_generator_inputs`? That's where we set the `streaming_callback` inside `generator_inputs` to pass to the ChatGenerator.